cirq-core/cirq/transformers/analytical_decompositions/quantum_shannon_decomposition.py

```diff
@@ -0,0 +1,247 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Utility methods for decomposing arbitrary n-qubit (2^n x 2^n) unitary."""
+
+from scipy.linalg import cossin
+
+import numpy as np
+
+import cirq
+
+
+def quantum_shannon_decomposition(qubits: list, U: np.ndarray, ops=None):
```

Comment:
Use lower case names for function arguments. 
```suggestion
def quantum_shannon_decomposition(qubits: list, u: np.ndarray, ops=None):
```